From ee8b1ec67a32fc4bb19aedba94593cb144cb870d Mon Sep 17 00:00:00 2001
From: Justin Tsai <justin@gmail.com>
Date: Wed, 10 May 2023 14:00:08 +0800
Subject: [PATCH 08/12] add config file and fix build single secureboot image
 issue. including: 1.dule image need ubi have 8M of space ,change ubinize
 script to create 8M ubi space. openwrt/scripts/ubinize-image.sh 2.secureboot
 single image doesn't build hash rootfs,and can't pass hash check when bootup.
 openwrt/target/linux/mediatek/image/mt7986.mk 3.add gm config file

---
 openwrt/.config                               | 62 ++++++++++++++-------------
 openwrt/scripts/ubinize-image.sh              |  9 +++-
 openwrt/target/linux/mediatek/image/mt7986.mk |  2 +-
 3 files changed, 40 insertions(+), 33 deletions(-)

diff --git a/openwrt/.config b/openwrt/.config
index 7eba717..5936b5c 100644
--- a/openwrt/.config
+++ b/openwrt/.config
@@ -86,8 +86,7 @@ CONFIG_TARGET_MULTI_PROFILE=y
 CONFIG_TARGET_PER_DEVICE_ROOTFS=y
 # CONFIG_TARGET_DEVICE_mediatek_mt7986_DEVICE_mt7986a-ax6000-2500wan-gsw-spim-nand-rfb is not set
 # CONFIG_TARGET_DEVICE_mediatek_mt7986_DEVICE_mt7986a-ax6000-2500wan-sd-rfb is not set
-CONFIG_TARGET_DEVICE_mediatek_mt7986_DEVICE_mt7986a-ax6000-2500wan-spim-nand-rfb=y
-CONFIG_TARGET_DEVICE_PACKAGES_mediatek_mt7986_DEVICE_mt7986a-ax6000-2500wan-spim-nand-rfb=""
+# CONFIG_TARGET_DEVICE_mediatek_mt7986_DEVICE_mt7986a-ax6000-2500wan-spim-nand-rfb is not set
 # CONFIG_TARGET_DEVICE_mediatek_mt7986_DEVICE_mt7986a-ax6000-2500wan-spim-nor-rfb is not set
 # CONFIG_TARGET_DEVICE_mediatek_mt7986_DEVICE_mt7986a-ax6000-emmc-rfb is not set
 # CONFIG_TARGET_DEVICE_mediatek_mt7986_DEVICE_mt7986a-ax6000-snfi-nand-rfb is not set
@@ -106,15 +105,12 @@ CONFIG_TARGET_DEVICE_PACKAGES_mediatek_mt7986_DEVICE_mt7986a-ax6000-spim-nand-rf
 # CONFIG_TARGET_DEVICE_mediatek_mt7986_DEVICE_mt7986b-ax6000-2500wan-gsw-spim-nand-rfb is not set
 # CONFIG_TARGET_DEVICE_mediatek_mt7986_DEVICE_mt7986b-ax6000-2500wan-sd-rfb is not set
 # CONFIG_TARGET_DEVICE_mediatek_mt7986_DEVICE_mt7986b-ax6000-2500wan-snfi-nand-rfb is not set
-CONFIG_TARGET_DEVICE_mediatek_mt7986_DEVICE_mt7986b-ax6000-2500wan-spim-nand-rfb=y
-CONFIG_TARGET_DEVICE_PACKAGES_mediatek_mt7986_DEVICE_mt7986b-ax6000-2500wan-spim-nand-rfb=""
+# CONFIG_TARGET_DEVICE_mediatek_mt7986_DEVICE_mt7986b-ax6000-2500wan-spim-nand-rfb is not set
 # CONFIG_TARGET_DEVICE_mediatek_mt7986_DEVICE_mt7986b-ax6000-2500wan-spim-nor-rfb is not set
 # CONFIG_TARGET_DEVICE_mediatek_mt7986_DEVICE_mt7986b-ax6000-emmc-rfb is not set
 # CONFIG_TARGET_DEVICE_mediatek_mt7986_DEVICE_mt7986b-ax6000-snfi-nand-rfb is not set
-CONFIG_TARGET_DEVICE_mediatek_mt7986_DEVICE_mt7986b-ax6000-spim-nand-rfb=y
-CONFIG_TARGET_DEVICE_PACKAGES_mediatek_mt7986_DEVICE_mt7986b-ax6000-spim-nand-rfb=""
-CONFIG_TARGET_DEVICE_mediatek_mt7986_DEVICE_mt7986b-ax6000-spim-nand-rfb-sb=y
-CONFIG_TARGET_DEVICE_PACKAGES_mediatek_mt7986_DEVICE_mt7986b-ax6000-spim-nand-rfb-sb=""
+# CONFIG_TARGET_DEVICE_mediatek_mt7986_DEVICE_mt7986b-ax6000-spim-nand-rfb is not set
+# CONFIG_TARGET_DEVICE_mediatek_mt7986_DEVICE_mt7986b-ax6000-spim-nand-rfb-sb is not set
 # CONFIG_TARGET_DEVICE_mediatek_mt7986_DEVICE_mt7986b-ax6000-spim-nor-rfb is not set
 # CONFIG_TARGET_DEVICE_mediatek_mt7986_DEVICE_mediatek_mt7986-fpga is not set
 # CONFIG_TARGET_DEVICE_mediatek_mt7986_DEVICE_mediatek_mt7986-fpga-ubi is not set
@@ -503,7 +499,6 @@ CONFIG_PER_FEED_REPO=y
 CONFIG_FEED_packages=y
 CONFIG_FEED_luci=y
 CONFIG_FEED_routing=y
-CONFIG_FEED_mtk_openwrt_feed=y
 
 #
 # Base system
@@ -1355,12 +1350,12 @@ CONFIG_BUSYBOX_DEFAULT_ROUTE=y
 # CONFIG_BUSYBOX_DEFAULT_FEATURE_TC_INGRESS is not set
 # CONFIG_BUSYBOX_DEFAULT_TCPSVD is not set
 # CONFIG_BUSYBOX_DEFAULT_UDPSVD is not set
-CONFIG_BUSYBOX_DEFAULT_TELNET=y
-CONFIG_BUSYBOX_DEFAULT_FEATURE_TELNET_TTYPE=y
-CONFIG_BUSYBOX_DEFAULT_FEATURE_TELNET_AUTOLOGIN=y
+# CONFIG_BUSYBOX_DEFAULT_TELNET is not set
+# CONFIG_BUSYBOX_DEFAULT_FEATURE_TELNET_TTYPE is not set
+# CONFIG_BUSYBOX_DEFAULT_FEATURE_TELNET_AUTOLOGIN is not set
 # CONFIG_BUSYBOX_DEFAULT_FEATURE_TELNET_WIDTH is not set
-CONFIG_BUSYBOX_DEFAULT_TELNETD=y
-CONFIG_BUSYBOX_DEFAULT_FEATURE_TELNETD_STANDALONE=y
+# CONFIG_BUSYBOX_DEFAULT_TELNETD is not set
+# CONFIG_BUSYBOX_DEFAULT_FEATURE_TELNETD_STANDALONE is not set
 # CONFIG_BUSYBOX_DEFAULT_FEATURE_TELNETD_INETD_WAIT is not set
 CONFIG_BUSYBOX_DEFAULT_TFTP=y
 # CONFIG_BUSYBOX_DEFAULT_FEATURE_TFTP_PROGRESS_BAR is not set
@@ -1576,14 +1571,14 @@ CONFIG_PACKAGE_ca-bundle=m
 CONFIG_PACKAGE_dnsmasq=y
 # CONFIG_PACKAGE_dnsmasq-dhcpv6 is not set
 # CONFIG_PACKAGE_dnsmasq-full is not set
-CONFIG_PACKAGE_dropbear=y
+CONFIG_PACKAGE_dropbear=m
 
 #
 # Configuration
 #
 CONFIG_DROPBEAR_CURVE25519=y
 # CONFIG_DROPBEAR_ECC is not set
-# CONFIG_DROPBEAR_ED25519 is not set
+CONFIG_DROPBEAR_ED25519=y
 CONFIG_DROPBEAR_CHACHA20POLY1305=y
 # CONFIG_DROPBEAR_ZLIB is not set
 CONFIG_DROPBEAR_DBCLIENT=y
@@ -1758,7 +1753,7 @@ CONFIG_ZABBIX_POSTGRESQL=y
 # CONFIG_PACKAGE_make is not set
 # CONFIG_PACKAGE_meson is not set
 CONFIG_PACKAGE_mt76-test=y
-CONFIG_PACKAGE_mt76-vendor=y
+# CONFIG_PACKAGE_mt76-vendor is not set
 # CONFIG_PACKAGE_mwctl is not set
 # CONFIG_PACKAGE_ninja is not set
 CONFIG_PACKAGE_objdump=y
@@ -2060,9 +2055,9 @@ CONFIG_PACKAGE_kmod-fs-autofs4=y
 # CONFIG_PACKAGE_kmod-fs-cifs is not set
 # CONFIG_PACKAGE_kmod-fs-configfs is not set
 # CONFIG_PACKAGE_kmod-fs-cramfs is not set
-# CONFIG_PACKAGE_kmod-fs-exfat is not set
+CONFIG_PACKAGE_kmod-fs-exfat=y
 CONFIG_PACKAGE_kmod-fs-exportfs=y
-# CONFIG_PACKAGE_kmod-fs-ext4 is not set
+CONFIG_PACKAGE_kmod-fs-ext4=y
 # CONFIG_PACKAGE_kmod-fs-f2fs is not set
 # CONFIG_PACKAGE_kmod-fs-fscache is not set
 # CONFIG_PACKAGE_kmod-fs-hfs is not set
@@ -2079,12 +2074,12 @@ CONFIG_PACKAGE_kmod-fs-nfs-common-rpcsec=y
 # CONFIG_PACKAGE_kmod-fs-nfs-v3 is not set
 # CONFIG_PACKAGE_kmod-fs-nfs-v4 is not set
 CONFIG_PACKAGE_kmod-fs-nfsd=y
-# CONFIG_PACKAGE_kmod-fs-ntfs is not set
+CONFIG_PACKAGE_kmod-fs-ntfs=y
 # CONFIG_PACKAGE_kmod-fs-reiserfs is not set
 # CONFIG_PACKAGE_kmod-fs-squashfs is not set
 # CONFIG_PACKAGE_kmod-fs-udf is not set
 CONFIG_PACKAGE_kmod-fs-vfat=y
-# CONFIG_PACKAGE_kmod-fs-xfs is not set
+CONFIG_PACKAGE_kmod-fs-xfs=y
 # CONFIG_PACKAGE_kmod-fuse is not set
 # end of Filesystems
 
@@ -2221,7 +2216,7 @@ CONFIG_PACKAGE_kmod-asn1-decoder=y
 # CONFIG_PACKAGE_kmod-lib-cordic is not set
 CONFIG_PACKAGE_kmod-lib-crc-ccitt=y
 # CONFIG_PACKAGE_kmod-lib-crc-itu-t is not set
-# CONFIG_PACKAGE_kmod-lib-crc16 is not set
+CONFIG_PACKAGE_kmod-lib-crc16=y
 CONFIG_PACKAGE_kmod-lib-crc32c=y
 # CONFIG_PACKAGE_kmod-lib-crc7 is not set
 # CONFIG_PACKAGE_kmod-lib-crc8 is not set
@@ -3744,7 +3739,7 @@ CONFIG_OPENSSL_ENGINE=y
 # CONFIG_OPENSSL_WITH_GOST is not set
 # CONFIG_PACKAGE_libopenssl-afalg is not set
 # CONFIG_PACKAGE_libopenssl-afalg_sync is not set
-# CONFIG_PACKAGE_libopenssl-conf is not set
+CONFIG_PACKAGE_libopenssl-conf=y
 # CONFIG_PACKAGE_libopenssl-devcrypto is not set
 CONFIG_PACKAGE_libwolfssl=m
 CONFIG_WOLFSSL_HAS_AES_CCM=y
@@ -4020,7 +4015,7 @@ CONFIG_PACKAGE_libpcap=y
 
 # CONFIG_PACKAGE_libpci is not set
 # CONFIG_PACKAGE_libpciaccess is not set
-# CONFIG_PACKAGE_libpcre is not set
+CONFIG_PACKAGE_libpcre=y
 # CONFIG_PACKAGE_libpcre16 is not set
 # CONFIG_PACKAGE_libpcre2 is not set
 # CONFIG_PACKAGE_libpcre2-16 is not set
@@ -4611,11 +4606,18 @@ CONFIG_POSTFIX_PCRE=y
 #
 CONFIG_PACKAGE_atenl=y
 CONFIG_PACKAGE_mii_mgr=y
+# CONFIG_PACKAGE_mtk-efuse-nl-tool is not set
 CONFIG_PACKAGE_regs=y
 CONFIG_PACKAGE_switch=y
 # end of Applications
 
 #
+# Drivers
+#
+# CONFIG_PACKAGE_kmod-mtk-efuse-nl-drv is not set
+# end of Drivers
+
+#
 # Misc
 #
 # CONFIG_PACKAGE_mtk_factory_rw is not set
@@ -5102,11 +5104,11 @@ CONFIG_PACKAGE_ip-tiny=y
 # SSH
 #
 # CONFIG_PACKAGE_autossh is not set
-# CONFIG_PACKAGE_openssh-client is not set
+CONFIG_PACKAGE_openssh-client=y
 # CONFIG_PACKAGE_openssh-client-utils is not set
-# CONFIG_PACKAGE_openssh-keygen is not set
+CONFIG_PACKAGE_openssh-keygen=y
 # CONFIG_PACKAGE_openssh-moduli is not set
-# CONFIG_PACKAGE_openssh-server is not set
+CONFIG_PACKAGE_openssh-server=y
 # CONFIG_PACKAGE_openssh-server-pam is not set
 # CONFIG_PACKAGE_openssh-sftp-avahi-service is not set
 # CONFIG_PACKAGE_openssh-sftp-client is not set
@@ -6242,7 +6244,7 @@ CONFIG_PACKAGE_libjson-script=y
 # CONFIG_PACKAGE_mpack is not set
 # CONFIG_PACKAGE_mt-st is not set
 # CONFIG_PACKAGE_namei is not set
-# CONFIG_PACKAGE_nand-utils is not set
+CONFIG_PACKAGE_nand-utils=y
 # CONFIG_PACKAGE_naywatch is not set
 # CONFIG_PACKAGE_netopeer2-cli is not set
 # CONFIG_PACKAGE_netopeer2-server is not set
@@ -6258,7 +6260,7 @@ CONFIG_PACKAGE_libjson-script=y
 # CONFIG_PACKAGE_openobex-apps is not set
 # CONFIG_PACKAGE_openocd is not set
 # CONFIG_PACKAGE_opensc-utils is not set
-# CONFIG_PACKAGE_openssl-util is not set
+CONFIG_PACKAGE_openssl-util=y
 # CONFIG_PACKAGE_openzwave is not set
 # CONFIG_PACKAGE_openzwave-config is not set
 # CONFIG_PACKAGE_owipcalc is not set
@@ -6331,7 +6333,7 @@ CONFIG_STRACE_NONE=y
 # CONFIG_PACKAGE_sysrepo is not set
 # CONFIG_PACKAGE_sysrepocfg is not set
 # CONFIG_PACKAGE_sysrepoctl is not set
-# CONFIG_PACKAGE_sysstat is not set
+CONFIG_PACKAGE_sysstat=y
 # CONFIG_PACKAGE_tar is not set
 # CONFIG_PACKAGE_taskwarrior is not set
 # CONFIG_PACKAGE_telegraf is not set
diff --git a/openwrt/scripts/ubinize-image.sh b/openwrt/scripts/ubinize-image.sh
index c6f8bce..23d2ee8 100755
--- a/openwrt/scripts/ubinize-image.sh
+++ b/openwrt/scripts/ubinize-image.sh
@@ -26,7 +26,12 @@ ubivol() {
 		echo "image=$image"
 		[ -n "$size" ] && echo "vol_size=${size}"
 	else
-		echo "vol_size=1MiB"
+		if [ "$size" ]; then
+			echo "vol_size=${size}"
+		else
+			echo "vol_size=1MiB"
+		fi
+
 	fi
 	if [ "$autoresize" ]; then
 		echo "vol_flags=autoresize"
@@ -79,7 +84,7 @@ ubilayout() {
 	ubivol $vol_id rootfs "$2" "$autoresize" "$rootsize"
 
 	vol_id=$(( $vol_id + 1 ))
-	[ "$rootfs_type" = "ubifs" ] || ubivol $vol_id rootfs_data "" 1
+	[ "$rootfs_type" = "ubifs" ] || ubivol $vol_id rootfs_data "" "" "8MiB"
 }
 
 set_ubinize_seq() {
diff --git a/openwrt/target/linux/mediatek/image/mt7986.mk b/openwrt/target/linux/mediatek/image/mt7986.mk
index b67cb33..7158d55 100644
--- a/openwrt/target/linux/mediatek/image/mt7986.mk
+++ b/openwrt/target/linux/mediatek/image/mt7986.mk
@@ -13,7 +13,7 @@ define Device/mt7986a-ax6000-spim-nand-rfb-sb
   IMAGE_SIZE := 65536k
   KERNEL_IN_UBI := 1
   IMAGES += factory.bin
-  IMAGE/factory.bin := append-ubi | check-size $$$$(IMAGE_SIZE)
+  IMAGE/factory.bin := append-ubi rootfs=$$$$(IMAGE_ROOTFS)-hashed | check-size $$$$(IMAGE_SIZE)
   IMAGE/sysupgrade.bin := sysupgrade-tar rootfs=$$$$(IMAGE_ROOTFS)-hashed | \
 	append-metadata
   FIT_KEY_DIR := $(TOPDIR)/../keys
-- 
2.7.4

